app/models/target.rb

```diff
@@ -27,4 +27,11 @@ class Target < ApplicationRecord
 
   validates :title, :longitude, :latitude, :radius, presence: true
   validates :radius, numericality: { only_integer: true }
+  validate :number_of_user_targets
+
+  private
+
+  def number_of_user_targets
+    errors.add(:user, 'exceeded number of targets limit') if user&.can_add_more_targets?
```

Comment:
```suggestion
    errors.add(:user, 'exceeded number of targets limit') if user.can_add_more_targets?
```
I think it's not necessary to add the safe navigation here